packages/x-data-grid-premium/src/components/GridAggregationRowOverlay.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import { forwardRef } from '@mui/x-internals/forwardRef';
+import { styled } from '@mui/system';
+import composeClasses from '@mui/utils/composeClasses';
+import { getDataGridUtilityClass, useGridRootProps } from '@mui/x-data-grid-pro';
+import { GridSkeletonLoadingOverlayInner, useGridSelector } from '@mui/x-data-grid-pro/internals';
+import { useGridApiContext } from '../hooks/utils/useGridApiContext';
+import { gridAggregationModelSelector } from '../hooks/features/aggregation/gridAggregationSelectors';
+import { DataGridPremiumProcessedProps } from '../models/dataGridPremiumProps';
+
+const GridAggregationRowOverlayWrapper = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'AggregationRowOverlayWrapper',
+})({
+  backgroundColor: 'var(--DataGrid-pinnedBackground)',
+});
+
+type OwnerState = { classes: DataGridPremiumProcessedProps['classes'] };
+
+const useUtilityClasses = (ownerState: OwnerState) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['aggregationRowOverlayWrapper'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const GridAggregationRowOverlay = forwardRef<HTMLDivElement, React.HTMLAttributes<HTMLDivElement>>(
+  function GridAggregationRowOverlay(props, forwardedRef) {
+    const apiRef = useGridApiContext();
+    const rootProps = useGridRootProps();
+    const classes = useUtilityClasses({ classes: rootProps.classes });
+    const aggregationModel = useGridSelector(apiRef, gridAggregationModelSelector);
+
+    const visibleColumns = new Set(Object.keys(aggregationModel));
+
+    return (
+      <GridAggregationRowOverlayWrapper className={classes.root}>
+        <GridSkeletonLoadingOverlayInner
+          {...props}
+          skeletonRowsCount={1}
+          visibleColumns={visibleColumns}
+          showFirstRowBorder
+          ref={forwardedRef}
+        />
```

Comment:
For the skeleton aggregation row, we also need to apply `backgroundColor: 'var(--DataGrid-pinnedBackground)'` so that the pinned row isn't transparent:

<img width="643" alt="Screenshot 2025-02-10 at 13 51 08" src="https://github.com/user-attachments/assets/e75de7dd-0ee8-455f-b686-c2cf1f4b3bb3" />

Reply:
I thought it's already [applied](https://github.com/MBilalShafi/mui-x/blob/863115a76152eb41f410ea7c121f0edbdda3de63/packages/x-data-grid-premium/src/components/GridAggregationRowOverlay.tsx#L15) 🤔 

Could you share which demo you can reproduce the issue on?

Reply:
Ah, I think you want to apply the style to `${gridClasses.row}` so that the background color takes up the width of all of the cells, similar to how we do here: https://github.com/mui/mui-x/blob/master/packages/x-data-grid/src/components/containers/GridRootStyles.ts#L600-L601

I found it by resizing a column on [this demo](https://deploy-preview-16282--material-ui-x.netlify.app/x/react-data-grid/server-side-data/aggregation/#usage-with-lazy-loading) to create an overflow.

Reply:
Nvm, I'm able to reproduce it, the issue is on the far right side of the row when there's a horizontal scroll. On it.